### PR TITLE
Add project item

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Depends on: https://github.com/toggle-corp/react-store/pull/748

### CODE REVIEW

File: src/views/Home/ProjectItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    }), []);
+
+    const convertedProjectActivity = useMemo(() => (
+        projectActivity?.map(pa => ({
+            count: pa.count,
+            date: (new Date(pa.date)).getTime(),
+        })).sort((a, b) => compareDate(a.date, b.date))
+    ), [projectActivity]);
+
+    console.warn('here', props);
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]: remove console.warn()

File: src/views/Home/ProjectItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    return (
+        <div className={_cs(className, styles.projectItem)}>
+            <header className={styles.header}>
+                <div className={styles.headerLeft}>
+                    <h2>
+                        {title}
+                    </h2>
+                    <div className={styles.dateContainer}>
+                        <FormattedDate
+                            title={_ts('home.recentProjects', 'startDateLabel')}
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx:136]: title props is not supported by FormattedDate. It leads to linting error.

File: src/views/Home/ProjectItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                            date={startDate}
+                            mode="MMM yyyy"
+                            emptyComponent={emptyComponent}
+                        />
+                        {startDate && endDate && (
+                            <div className={styles.separator}>-</div>
+                        )}
+                        <FormattedDate
+                            title={_ts('home.recentProjects', 'endDateLabel')}
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx:145]: same as above

File: src/views/Home/ProjectItem/index.tsx [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+        <div className={_cs(className, styles.projectItem)}>
+            <header className={styles.header}>
+                <div className={styles.headerLeft}>
+                    <h2>
+                        {title}
+                    </h2>
+                    <div className={styles.dateContainer}>
+                        <FormattedDate
+                            title={_ts('home.recentProjects', 'startDateLabel')}
+                            date={startDate}
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]:
```suggestion
                            value={startDate}
```

File: src/views/Home/ProjectItem/index.tsx [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+                            mode="MMM yyyy"
+                            emptyComponent={emptyComponent}
+                        />
+                        {startDate && endDate && (
+                            <div className={styles.separator}>-</div>
+                        )}
+                        <FormattedDate
+                            title={_ts('home.recentProjects', 'endDateLabel')}
+                            date={endDate}
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]:
```suggestion
                            value={endDate}
```

File: src/views/Home/ProjectItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+                    <div className={styles.rightTop}>
+                        <InformationBox
+                            className={styles.infoBox}
+                            icon={(
+                                <Icon
+                                    className={styles.icon}
+                                    name="bookmarkIcon"
+                                />
+                            )}
+                            label="Total Sources"
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]: I think we should use translations here.

File: src/views/Home/ProjectItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+                            value={totalSources}
+                            variant="negativeAccent"
+                        />
+                        <div className={styles.progressBars}>
+                            <ProgressLine
+                                className={styles.progressBar}
+                                progress={(totalSourcesValidated / totalSources) * 100}
+                                title="Sources Tagged & Validated"
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]: I think we should use translations here

File: src/views/Home/ProjectItem/index.tsx [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+                                variant="complement1"
+                            />
+                            <ProgressLine
+                                className={styles.progressBar}
+                                progress={(totalSourcesTagged / totalSources) * 100}
+                                title="Sources Tagged"
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]: use of translations is recommended

File: src/views/Home/ProjectItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+                                variant="complement2"
+                            />
+                            <ProgressLine
+                                className={styles.progressBar}
+                                progress={
+                                    ((totalSources - totalSourcesTagged) / totalSources) * 100
+                                }
+                                title="Sources Untagged"
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]: use of translations is recommended

File: src/components/viz/ProgressLine/index.tsx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+function ProgressLine(props: Props) {
+    const {
+        className,
+        progressClassName,
+        progress,
+        title,
+        variant = 'complement1',
+    } = props;
+
+    const progressWidth = `${bound(progress, 0, 100)}%`;
```
Comment (samshara) [inline src/components/viz/ProgressLine/index.tsx]: does this function properly handle when progress is NaN ?.  In the usage below when the totalSources is 0, the progress bar is of full width. Is this the behavior we want?

File: src/views/Home/ProjectItem/index.tsx [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+
+    return (
+        <div className={_cs(className, styles.recentlyActiveUser)}>
+            <div className={styles.name}>
+                {name}
+            </div>
+            <FormattedDate
+                className={styles.date}
+                // FIXME: Remove this fallback
+                date={date ?? Date.now()}
```
Comment (samshara) [inline src/views/Home/ProjectItem/index.tsx]:
```suggestion
                value={date ?? Date.now()}
```

File: src/views/Home/index.tsx [hunk 12]
```diff
@@ -113,9 +211,12 @@ function Home(props: ViewProps) {
...
                             </header>
-                            <RecentProjects
-                                className={styles.recentProjects}
-                                selectedProject={selectedProject}
+                            <ListView
+                                data={finalRecentProjects}
+                                rendererParams={recentProjectsRendererParams}
+                                renderer={ProjectItem}
+                                keySelector={recentProjectKeySelector}
+                                pending={projectDetailsPending || projectStatsPending}
```
Comment (samshara) [inline src/views/Home/index.tsx:219]: why are we using ListView to render a singular recent project?

Comment (AdityaKhatri) [reply src/views/Home/index.tsx:219]: We'll later use the recent projects API to show 3 projects.

File: src/views/Home/ProjectItem/index.tsx (hunk 2 above)
Comment (AdityaKhatri) [reply src/views/Home/ProjectItem/index.tsx:136]: I forgot to attach the respective PR in react-store

File: src/views/Home/ProjectItem/index.tsx (hunk 3 above)
Comment (AdityaKhatri) [reply src/views/Home/ProjectItem/index.tsx:145]: I forgot to attach the respective PR in react-store

**Pull request merged**